nuleaf-source/database/posts.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Posts database access object. Opinionated wrapper around mongoose.
+ *
+ * Handles requests to posts collection.
+ */
+
+/* jslint node: true */
+'use strict';
+
+var Helper = require('../util/helper');
+
+var Post = require('../models/post');
+
+
+/**
+ * Finds posts matching conditions and returns a collection of posts.
+ * @param {Object} Conditions:
+ *                   name            : Filter for posts with this name.
+ *                   content         : Filter for posts with this content.
+ *                   author          : Filter for posts with this author.
+ *                   sort            : Stores a number that determines if the results are shown in de/ascending order.
+ *                   sortBy          : Stores the attribute above that the results are sorted by.
+ *                   skip            : Return a certain number of results after a certain number of documents.
+ *                   limit           : Used to specify the maximum number of results to be returned.
+ *                   created_before  : Filter for posts after this date.
+ *                   created_after   : Filter for posts before this date.
+ *                   published_before: Filter for posts after this date.
+ *                   published_after : Filter for posts before this date.
+ *                   modified_before : Filter for posts after this date.
+ *                   modified_after  : Filter for posts before this date.
+ *
+ * @return {Error}, {Array} Array of posts, or empty if none found matching conditions.
+ */
+exports.find = function(conditions, callback) {
+  if (typeof conditions === 'function') {
+    callback = conditions;
+    conditions = {};
+
+    var skip = +conditions.skip || 0;
+    var limit = +conditions.limit || 0;
+
+    var _condititons = buildConditions(conditions);
+    var query = Post.find(_conditions).skip(skip).limit(limit);
+  }
```

Comment:
Skip, limit, _conditions, and query should be outside of the if statement scope.